pkg/ingestor/parser/csaf/parser_csaf.go

```diff
@@ -221,43 +273,42 @@ func (c *csafParser) generateVexIngest(ctx context.Context, vulnInput *generated
 	return vi
 }
 
+// GetPredicates generates the VEX and CertifyVuln predicates for the CSAF document.
+//
+// It returns a pointer to an assembler.IngestPredicates struct containing the
+// generated VEX and CertifyVuln predicates.
 func (c *csafParser) GetPredicates(ctx context.Context) *assembler.IngestPredicates {
 	rv := &assembler.IngestPredicates{}
 	var vis []assembler.VexIngest
 	var cvs []assembler.CertifyVulnIngest
 
-	if len(c.csaf.Vulnerabilities) > 0 {
+	for _, v := range c.csaf.Vulnerabilities {
+		vuln, err := helpers.CreateVulnInput(v.CVE)
+		if err != nil {
+			return nil
+		}
 
-		for _, v := range c.csaf.Vulnerabilities {
-			vuln, err := helpers.CreateVulnInput(v.CVE)
-			if err != nil {
-				return nil
-			}
+		statuses := []string{"fixed", "known_not_affected", "known_affected", "first_affected", "first_fixed", "last_affected", "recommended", "under_investigation"}
+		for _, status := range statuses {
+			products := v.ProductStatus[status]
+			for _, product := range products {
+				vi := c.generateVexIngest(ctx, vuln, &v, status, product)
+				if vi == nil {
+					continue
+				}
 
-			statuses := []string{"fixed", "known_not_affected", "known_affected", "first_affected", "first_fixed", "last_affected", "recommended", "under_investigation"}
-			for _, status := range statuses {
-				products := v.ProductStatus[status]
-				if len(products) > 0 {
-					for _, product := range products {
```

Comment:
Removed redundant check. The `for` loop accomplishes what the `if len(products) > 0` is supposed to.